src/main/java/com/jagrosh/jmusicbot/commands/dj/PlaynowCmd.java

```diff
@@ -0,0 +1,131 @@
+    }
+    
+    private class ResultHandler implements AudioLoadResultHandler
+    {
+        private final Message m;
+        private final CommandEvent event;
+        private final boolean ytsearch;
+        
+        private ResultHandler(Message m, CommandEvent event, boolean ytsearch)
+        {
+            this.m = m;
+            this.event = event;
+            this.ytsearch = ytsearch;
+        }
+        
+        private void loadSingle(AudioTrack track)
+        {
+            if(bot.getConfig().isTooLong(track))
+            {
+                m.editMessage(FormatUtil.filter(event.getClient().getWarning()+" This track (**"+track.getInfo().title+"**) is longer than the allowed maximum: `"
+                        +FormatUtil.formatTime(track.getDuration())+"` > `"+FormatUtil.formatTime(bot.getConfig().getMaxSeconds()*1000)+"`")).queue();
+                return;
+            }
+            AudioHandler handler = (AudioHandler)event.getGuild().getAudioManager().getSendingHandler();
+            handler.addTrackToFront(new QueuedTrack(track, event.getAuthor()));
+            String addMsg = FormatUtil.filter(event.getClient().getSuccess()+" Added **"+track.getInfo().title
+                    +"** (`"+FormatUtil.formatTime(track.getDuration())+"`) to begin playing");
+            m.editMessage(addMsg).queue();
+            handler.getPlayer().stopTrack();
+        }
+        
+        @Override
+        public void trackLoaded(AudioTrack track)
+        {
+            loadSingle(track);
+        }
+
+        @Override
+        public void playlistLoaded(AudioPlaylist playlist)
+        {
+            AudioTrack single;
+            if(playlist.getTracks().size()==1 || playlist.isSearchResult())
+                single = playlist.getSelectedTrack()==null ? playlist.getTracks().get(0) : playlist.getSelectedTrack();
+            else if (playlist.getSelectedTrack()!=null)
+                single = playlist.getSelectedTrack();
+            else
+                single = playlist.getTracks().get(0);
+            loadSingle(single);
+        }
+
+        @Override
+        public void noMatches()
+        {
+            if(ytsearch)
+                m.editMessage(FormatUtil.filter(event.getClient().getWarning()+" No results found for `"+event.getArgs()+"`.")).queue();
+            else
+                bot.getPlayerManager().loadItemOrdered(event.getGuild(), "ytsearch:"+event.getArgs(), new ResultHandler(m,event,true));
+        }
+
+        @Override
+        public void loadFailed(FriendlyException throwable)
+        {
+            if(throwable.severity==FriendlyException.Severity.COMMON)
+                m.editMessage(event.getClient().getError()+" Error loading: "+throwable.getMessage()).queue();
+            else
+                m.editMessage(event.getClient().getError()+" Error loading track.").queue();
+        }
+    }
```

Comment:
I feel like this is a redundant class that could be moved out into another class where it will be inherted to the PlayCmd and your Play**N**owCmd

Reply:
You're entirely right, it's just a lot of duplicated code. I'll get to fixing that :)

Reply:
As for not using an uppercase for `Now` in the class or file name, that's just made with the same convention as the rest of the project